src/GafferScene/SceneNode.cpp

```diff
@@ -384,66 +406,42 @@ IECore::ConstPathMatcherDataPtr SceneNode::computeSet( const IECore::InternedStr
+Gaffer::ValuePlug::CachePolicy SceneNode::hashCachePolicy( const Gaffer::ValuePlug *output ) const
 {
-	ConstInternedStringVectorDataPtr computedChildNames;
-	if( !childNamesData )
-	{
-		computedChildNames = out->childNames( path );
-		childNamesData = computedChildNames.get();
-	}
-	const vector<InternedString> &childNames = childNamesData->readable();
-
-	IECore::MurmurHash result;
-	if( childNames.size() )
+	if( auto parent = output->parent<ScenePlug>() )
 	{
-		ScenePlug::PathScope pathScope( Context::current() );
-
-		ScenePath childPath( path );
-		childPath.push_back( InternedString() ); // room for the child name
-		for( vector<InternedString>::const_iterator it = childNames.begin(); it != childNames.end(); it++ )
+		if( output == parent->childBoundsPlug() )
 		{
-			childPath[path.size()] = *it;
-			pathScope.setPath( childPath );
-			out->boundPlug()->hash( result );
-			out->transformPlug()->hash( result );
+			return ValuePlug::CachePolicy::TaskIsolation;
 		}
 	}
-	else
-	{
-		result.append( typeId() );
-		result.append( "emptyBound" );
-	}
-	return result;
+
+	return ComputeNode::hashCachePolicy( output );
 }
 
-Imath::Box3f SceneNode::unionOfTransformedChildBounds( const ScenePath &path, const ScenePlug *out, const IECore::InternedStringVectorData *childNamesData ) const
+Gaffer::ValuePlug::CachePolicy SceneNode::computeCachePolicy( const Gaffer::ValuePlug *output ) const
 {
-	ConstInternedStringVectorDataPtr computedChildNames;
-	if( !childNamesData )
-	{
-		computedChildNames = out->childNames( path );
-		childNamesData = computedChildNames.get();
-	}
-	const vector<InternedString> &childNames = childNamesData->readable();
-
-	Box3f result;
-	if( childNames.size() )
+	if( auto parent = output->parent<ScenePlug>() )
 	{
-		ScenePlug::PathScope pathScope( Context::current() );
-
-		ScenePath childPath( path );
-		childPath.push_back( InternedString() ); // room for the child name
-		for( vector<InternedString>::const_iterator it = childNames.begin(); it != childNames.end(); it++ )
+		if( output == parent->childBoundsPlug() )
 		{
-			childPath[path.size()] = *it;
-			pathScope.setPath( childPath );
-			Box3f childBound = out->boundPlug()->getValue();
-			childBound = transform( childBound, out->transformPlug()->getValue() );
-			result.extendBy( childBound );
+			return ValuePlug::CachePolicy::TaskIsolation;
 		}
 	}
-	return result;
+
+	return ComputeNode::computeCachePolicy( output );
+}
+
+IECore::MurmurHash SceneNode::hashOfTransformedChildBounds( const ScenePath &path, const ScenePlug *out, const IECore::InternedStringVectorData *childNamesData ) const
+{
+	ScenePlug::PathScope pathScope( Context::current(), path );
+	return out->childBoundsPlug()->hash();
+}
+
+Imath::Box3f SceneNode::unionOfTransformedChildBounds( const ScenePath &path, const ScenePlug *out, const IECore::InternedStringVectorData *childNamesData ) const
```

Comment:
This naming convention is now looking pretty inconsistent with the rest of Gaffer.

Usually, the method name and plug name would match ( if the word transformed is actually important to understanding the function, it should be in the plug name ), and usually we would name it "[transformed]childBoundsHash", not "hashOfChildBounds"?

I guess it's called enough places that you don't want to change it, but would it be worth providing an optional new name and labelling the old inconsistent name as deprecated?

Reply:
Yes, I deprecated it in 9950141 already. The new name is `ScenePlug::childBounds()` as documented in Changes.md in that same commit.